### PR TITLE
Proposition interactions

### PR DESCRIPTION
**Pull request by sbenedicadb**
properly construct proposition interaction edge events and lots of tests

### CODE REVIEW

File: AEPMessaging/Tests/UnitTests/Messaging+EdgeEventsTests.swift [hunk 1]
```diff
@@ -427,117 +427,209 @@ class MessagingEdgeEventsTests: XCTestCase {
...
-        XCTAssertEqual(1, mockRuntime.dispatchedEvents.count)
-        let dispatchedEvent = mockRuntime.firstEvent
-        // validate no mixin in xdm map
-        let dispatchedEventData = dispatchedEvent?.data
-        let dispatchedXdmMap = dispatchedEventData?[MessagingConstants.XDM.Key.XDM] as? [String: Any]
-        XCTAssertNil(dispatchedXdmMap?[MessagingConstants.XDM.IAM.IN_APP_MIXIN_NAME])
+        XCTAssertEqual(0, mockRuntime.dispatchedEvents.count)
+    }
+    
+    func testSendPropositionInteractionInteractNoCharacteristics() throws {
```
Comment (rymorale) [inline AEPMessaging/Tests/UnitTests/Messaging+EdgeEventsTests.swift:619]: nit: is there a clearer way to name this test?

**Review by rymorale — APPROVED** [review]
looks good overall, just had a couple nits to take a look at.

File: AEPMessaging/Sources/Messaging+EdgeEvents.swift [hunk 2]
```diff
@@ -240,163 +240,106 @@ extension Messaging {
...
-                MessagingConstants.XDM.IAM.ACTION: interaction
+            let inappMessageTracking = [
+                MessagingConstants.XDM.IAM.Key.ACTION: interaction
             ]
-            xdmMap[MessagingConstants.XDM.IAM.IN_APP_MIXIN_NAME] = actionDict
+            let cjmXdm = [
+                MessagingConstants.XDM.IAM.Key.IN_APP_MESSAGE_TRACKING: inappMessageTracking
+            ]
+            let encoder = JSONEncoder()
+            guard let cjmXdmData = try? encoder.encode(cjmXdm) else { return }
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply AEPMessaging/Sources/Messaging+EdgeEvents.swift:302]: this code will be changing soon - ajo team is changing the format of this request and this code will no longer be needed

File: AEPMessaging/Tests/UnitTests/Messaging+EdgeEventsTests.swift (hunk 1 above)
Comment (sbenedicadb) [reply AEPMessaging/Tests/UnitTests/Messaging+EdgeEventsTests.swift:619]: probably - but this test will be going away with the same update mentioned above :)

**Pull request merged**